### PR TITLE
[driver][test] Only check for unused plugin options

### PR DESCRIPTION
**Pull request by jakeegan**
This fixes matching `clang: error: argument unused during compilation: '-Werror' [-Werror,-Wunused-command-line-argument]` on AIX.

### CODE REVIEW

File: clang/test/Driver/plugin-driver-args.cpp [hunk 1]
```diff
@@ -23,5 +23,5 @@
 
 // Plugins are only relevant for the -cc1 phase. No warning should be raised
 // when only using the assembler. See GH #88173.
-// RUN: %clang -c -fpass-plugin=bar.so -fplugin=bar.so -fplugin-arg-bar-option -Werror -x assembler %s -### 2>&1 | FileCheck %s --check-prefix=CHECK-PLUGIN-ASM
-// CHECK-PLUGIN-ASM-NOT: argument unused during compilation
+// RUN: %clang -c -fpass-plugin=bar.so -fplugin=bar.so -fplugin-arg-bar-option -Wunused-command-line-argument -x assembler %s -### 2>&1 | FileCheck %s --check-prefix=CHECK-PLUGIN-ASM
+// CHECK-PLUGIN-ASM-NOT: argument unused during compilation: '-f{{[a-z-]*-plugin[^']*}}'
```
Comment (hubert-reinterpretcast) [inline clang/test/Driver/plugin-driver-args.cpp:27]: Sorry for my mistake "offline"; we want to catch `-fplugin*` too:
```suggestion
// CHECK-PLUGIN-ASM-NOT: argument unused during compilation: '-f{{[a-z-]*plugin[^']*}}'
```

**Review by hubert-reinterpretcast — APPROVED** [review]
LGTM!

**Review by MaskRay — APPROVED** [review]
Consider use a test tag, e.g. `[Driver,test]`

**Comment by jakeegan** [comment]
CI failure doesn't seem to be related

**Pull request merged**